### DISCUSSION TITLE
Unable to automate: Data-importer doesn't remember which accounts spectre accounts map to.

### DISCUSSION THREAD

**Discussion by demiGod095** [firefly-iii/firefly-iii#9067] [Firefly III] [answered]
Hi everyone,
I am having some trouble with the data-importer. I am on **Data Importer, v1.5.2** 
I can see from the json that its generated an account mapping from Spectre as follows:
```
    "accounts": {
        "xxxxxxxxx": 3,
        "yyyyyyyyy": 1
    },
``` 
However, when I run the importer again, select the same json from the dropdown, it shows me the following:
![image](https://github.com/user-attachments/assets/776b75a9-f87a-4faa-8bce-7143d9bd346f)
Neither of these `xxx` and `yyy` accounts are ANZ CC, ANZ CC in fact has an ID of `10` in firefly-3 (ANZ CC's URL looks like: [https://ff3.domain.com/accounts/show/10](https://ff3.domain.com/accounts/show/10))

I went throught the issues and discussions and didn't find anyone with the same issue, Please let me know if this has been answered elsewhere.
Please also let me know if there's any other information you need for debugging this.
Thank you!

**Comment by JC5** [comment]
After you have corrected the import, redownload the configuration file and use the new version instead.

**Reply by demiGod095** [reply]
@JC5 thanks for the suggestion. 
I am getting the same behaviour regardless of how many times I run the data import after fixing the import..
I have tried creating new import.json from scratch and get the same behaviour on the next run.

**Reply by JC5** [reply]
No you're right, that wouldn't have worked. I thought your import configuration had old id's in it. But alas, this is a bug. It will be fixed in the next release of the data importer :)

**Reply by demiGod095** [reply]
Sweet thanks! 
Any easy way to patch my install for the time being?

**Reply by JC5** [reply, marked as answer]
v1.5.3 should be out already 😁

**Reply by demiGod095** [reply]
Thank you.
I can confirm that my previously generated configs are working without changes now.
